commit 56fbd9070447e1a02dcd0785d420b5aecde7f97e
Author: daitamae <dai.tamae@gmail.com>
Date:   Mon Dec 5 20:09:55 2022 -0300

    Correct api service for deploy

diff --git a/.env b/.env
index 4d1571d..3462d2e 100644
--- a/.env
+++ b/.env
@@ -10,17 +10,15 @@ NATS_INNER_PORT_1=4222
 NATS_INNER_PORT_2=8222
 MLFLOW_DB_INNER_PORT=3306
 MLFLOW_INNER_PORT=5000
-MONGO_INNER_PORT=27017
-
-API_PORT=3000
-ME_PORT=8081
-NATS_PORT_1=4222
-NATS_PORT_2=8222
-MINIO_PORT_1=9000
-MINIO_PORT_2=9001
-MLFLOW_DB_PORT=3306
-MLFLOW_PORT=5000
 
+API_PORT=28000
+ME_PORT=28001
+NATS_PORT_1=8422
+NATS_PORT_2=8822
+MINIO_PORT_1=8900
+MINIO_PORT_2=8901
+MLFLOW_DB_PORT=8336
+MLFLOW_PORT=8500
 
 MLFLOW_S3_ENDPOINT_URL=http://minio:${MINIO_INNER_PORT_1}
 AWS_ACCESS_KEY_ID=minio
@@ -41,7 +39,7 @@ MYSQL_USER=mlflow_user
 MYSQL_PASSWORD=mlflow
 MYSQL_ROOT_PASSWORD=mysql
 
-COMPOSE_PROJECT_NAME=spira
+COMPOSE_PROJECT_NAME=spira-inference
 
 ### API ENVIRONMENT VARIABLES
 
@@ -53,7 +51,7 @@ deprecated=auto
 
 MONGO_INITDB_ROOT_USERNAME=username
 MONGO_INITDB_ROOT_PASSWORD=password
-mongo_conn_url=mongodb://${MONGO_INITDB_ROOT_USERNAME}:${MONGO_INITDB_ROOT_PASSWORD}@mongo:${MONGO_INNER_PORT}/
+mongo_conn_url=mongodb://${MONGO_INITDB_ROOT_USERNAME}:${MONGO_INITDB_ROOT_PASSWORD}@mongo:27017/
 database_name=spira_db
 user_collection_name=users
 model_collection_name=models
@@ -79,4 +77,4 @@ receiving_channel=fake_channel_2
 mlflow_conn_url=http://mlflow:${MLFLOW_INNER_PORT}
 model_path=models:/pyfunc-test-model/Staging
 
-default_extension=${file_extension}
\ No newline at end of file
+default_extension=${file_extension}
diff --git a/build-api.sh b/build-api.sh
index 0330152..b07c88d 100644
--- a/build-api.sh
+++ b/build-api.sh
@@ -1,3 +1,2 @@
 #!/bin/bash
-chmod +x ./wait-for-it.sh
-docker-compose -f docker-compose.api.yml up --build api mongo mongo-express nats minio mlflow-mc mlflow db
\ No newline at end of file
+docker compose -f docker-compose.api.yml up --build api mongo mongo-express nats minio mlflow-mc mlflow db
diff --git a/docker-compose-api@.service b/docker-compose-api@.service
index 99b430f..fe69ba0 100644
--- a/docker-compose-api@.service
+++ b/docker-compose-api@.service
@@ -4,13 +4,13 @@ PartOf=docker.service
 After=docker.service
 
 [Service]
-Type=oneshot
-WorkingDirectory=/path-to-project/
-ExecStart=/usr/bin/bash build-api.sh
-ExecStop=/usr/bin/bash stop-api.sh
+Type=simple
+WorkingDirectory=/home/spira-inference-system/deploy/
+ExecStart=/bin/bash build-api.sh
+ExecStop=/bin/bash stop-api.sh
 StandardOutput=syslog
 StandardError=syslog
 SyslogIdentifier=SPIRA-INFERENCE
 
 [Install] 
-WantedBy=multi-user.target
\ No newline at end of file
+WantedBy=multi-user.target
diff --git a/docker-compose.api.yml b/docker-compose.api.yml
index 28da01d..f925cb4 100644
--- a/docker-compose.api.yml
+++ b/docker-compose.api.yml
@@ -75,6 +75,9 @@ services:
   mlflow:
     image: ${MLFLOW_SERVER_IMAGE_NAME}
     restart: always
+    depends_on: 
+      - db
+      - minio
     ports:
       - ${MLFLOW_PORT}:${MLFLOW_INNER_PORT}
     env_file: 
@@ -85,4 +88,4 @@ volumes:
   dbdata:
   minio_data:
 
-  
\ No newline at end of file
+  
